### PR TITLE
MueLu/Ifpack2:  Pushing BlockCrsMatrix Smoother support to coarse levels

### PR DESCRIPTION
**Pull request by csiefer2**
TESTS: RHEL7 (including two new ones)

Notes for reviewers:  BlockCrsMatrix is for small blocks (e.g. 3x3 elasticity), rather than BlockedMatrix, which is for big blocks (e.g. velocity/pressure for Navier-Stokes).

### CODE REVIEW

**Review by rstumin — DISMISSED** [review]
Looks good.

**Review by jhux2 — APPROVED** [review]
lgtm

File: packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp [hunk 1]
```diff
@@ -886,6 +834,68 @@ namespace MueLuTests {
...
+         }
+         RCP<Map> new_map = Xpetra::MapFactory<LO,GO,NO>::Build(lib,GO_INVALID,owned_rows(),0,comm);
+         if(new_map.is_null()) throw std::runtime_error("BuildBlockMatrixAsPoint: Map constructor failed");
+
+
+         // Block Graph / Matrix
+         RCP<CrsMatrix> new_matrix = Xpetra::CrsMatrixFactory<SC,LO,GO,NO>::Build(new_map,blocksize*old_graph->getNodeMaxNumRowEntries());
+         if(new_matrix.is_null()) throw std::runtime_error("BuildBlockMatrixAsPoint: Matrix constructor failed");
+         for(LO i=0; i<orig_num_rows; i++) {
+           Teuchos::ArrayView<const GO> old_indices;
```
Comment (mayrmt) [inline packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp:874]:
```suggestion
           Teuchos::ArrayView<const LO> old_indices;
```

File: packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp [hunk 2]
```diff
@@ -886,6 +834,68 @@ namespace MueLuTests {
...
+           Teuchos::ArrayView<const SC> old_values;
+           Teuchos::Array<GO> new_indices(1);
+           Teuchos::Array<SC> new_values(1);
+           old_matrix->getLocalRowView(i,old_indices,old_values);
+           for(int ii=0; ii<blocksize; ii++) {           
+             GO GRID = new_map->getGlobalElement(i*blocksize+ii);
+             for(LO j=0; j<(LO)old_indices.size(); j++) {
+               for(int jj=0; jj<blocksize; jj++) {           
+                new_indices[0] = old_colmap->getGlobalElement(old_indices[j]) * blocksize + jj;
+                new_values[0]  = old_values[j] * ( (ii == jj && i == old_indices[j] ) ? blocksize*blocksize : 1 );
```
Comment (mayrmt) [inline packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp:884]:
```suggestion
                new_values[0]  = old_values[j] * static_cast<SC>( (ii == jj && i == old_indices[j] ) ? blocksize*blocksize : 1 ));
```

**Review by mayrmt — COMMENTED** [review]
@csiefer2 I highlighted two changes, that I had to do to get this branch compiling, while looking into #8770.

File: packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp]: Thanks!

File: packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/test/unit_tests/MueLu_TestHelpers.hpp]: Fixed!

**Review by cgcgcg — APPROVED** [review]

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**